plugins/woocommerce-admin/client/marketplace/components/search/search.tsx

```diff
@@ -1,12 +1,94 @@
 /**
  * External dependencies
  */
+import { __ } from '@wordpress/i18n';
+import { Icon, search } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import './search.scss';
 
-export default function Search() {
-	return <div className="woocommerce-marketplace__search">SEARCH 🔎</div>;
+const searchPlaceholder = __( 'Search extensions and themes', 'woocommerce' );
+
+const marketplaceAPI = 'https://woocommerce.com/wp-json';
+
+export interface SearchProps {
+	locale?: string | 'en_US';
+	country?: string | undefined;
 }
+
+/**
+ * Search component.
+ *
+ * @param {SearchProps} props - Search props: locale and country.
+ * @return {JSX.Element} Search component.
+ */
+function Search( props: SearchProps ): JSX.Element {
+	const locale = props.locale ?? 'en_US';
+	const country = props.country ?? '';
+
+	const build_parameter_string = (
+		query_string: string,
+		query_country: string,
+		query_locale: string
+	): string => {
+		const params = new URLSearchParams();
+		params.append( 'term', query_string );
+		params.append( 'country', query_country );
+		params.append( 'locale', query_locale );
+		return params.toString();
+	};
+
+	const runSearch = () => {
+		const query = (
+			document.getElementById( 'search-query' ) as HTMLInputElement
+		 ).value.trim();
+
+		const params = build_parameter_string( query, country, locale );
+		fetch( marketplaceAPI + '?' + params, {
+			method: 'GET',
+		} )
+			.then( ( response ) => response.json() )
+			.then( ( response ) => {
+				return response;
+			} );
+		return [];
+	};
+
+	const handleKeyUp = ( event: { key: string } ) => {
+		if ( event.key === 'Enter' ) {
+			runSearch();
+		}
+	};
+
+	const renderSearch = () => {
+		return (
+			<div className="marketplace-search-form">
```

Comment:
_Probably_ not worth changing, but if this was a `<form>` instead of a `<div>`:

1. You'd get the `onSubmit` event for free (which would be automatically be triggered by clicking the `<button>` or by pressing enter in the `<input>`, so you wouldn't need to handle `onKeyUp` on the `<input>` (checking for enter and calling `runSearch`): you could just `runSearch` when the form submits.
2. You could set the action="" of the `<form>` to the WCCOM version of the search. This way, the search would work even if JavaScript failed. Alternatively, it could have an action="" that submitted to the same page and capture the request that way, but it's much more work!
3. `<form>`s act as a landmark for screen readers, so they can jump to it. (You can, and probably should, "fake" this by adding `role="form"` to your `<div>` if you're keeping it as a `<div>`.)

Again: this probably isn't worth changing, just wanted to put the note here for educational purposes!

Reply:
This is such great info as always! I don't know that I'd have realised all this myself. Definitely one to note for future!